src/wallet/transaction.h

```diff
@@ -232,8 +215,10 @@ class CWalletTx
             setConfirmed();
         }
 
-        ReadOrderPos(nOrderPos, mapValue);
-        nTimeSmart = mapValue.count("timesmart") ? (unsigned int)atoi64(mapValue["timesmart"]) : 0;
+        const auto it_op = mapValue.find("n");
+        nOrderPos = (it_op != mapValue.end()) ? atoi64(it_op->second) : -1;
```

Comment:
At some point we'll want to replace the use of `atoi64`'s here, which is one of the last remaining uses in non-test code, with `ParseInt64` (as suggested by the developer notes).